src/finch/interface/lazy.py

```diff
@@ -327,7 +337,173 @@ def prod(arr: LazyTensor, dims) -> LazyTensor:
 
 def multiply(x1: LazyTensor, x2: LazyTensor) -> LazyTensor:
-    return elementwise(operator.mul, x1, x2)
\ No newline at end of file
+    return elementwise(operator.mul, x1, x2)
+
+def sum(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        dtype = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Calculates the sum of input array ``arr`` elements.
+    """
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+def any(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Test whether any element of input array ``arr`` along given axis is True.
+    """
+    return reduce(operator.or_, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+def all(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Test whether all elements of input array ``arr`` along given axis are True.
+    """
+    return reduce(operator.and_, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+# def _flatten(x):
+#     for y in x:
+#         if isinstance(y, (list, tuple)):
+#             yield from _flatten(y)
+#         else:
+#             yield y
+
+def min(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False    
+    ) -> LazyTensor:
+    """
+    Return the minimum of input array ``arr`` along given axis.
+    """
+#     num_rows = arr.shape[0]
+#     num_cols = arr.shape[1]
+
+#     if axis is None:
+#         return min(_flatten(arr))
+
+#     # column‑wise minimum
+#     if axis == 0:
+#         return [min(arr[r][c] for r in range(num_rows)) for c in range(num_cols)]
+#     # row‑wise minimum
+
+#     if axis == 1:
+#         return [min(row) for row in arr]
+
+    # TODO: Use operator.min to reduce?
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+def max(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Return the maximum of input array ``arr`` along given axis.
+    """
+    # TODO: Use operator.max to reduce?
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
+
+
+def mean(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Calculates the arithmetic mean of the input array ``arr``.
+    """
+    origin = np.asarray(arr.shape)
+    ele_no = origin[axis]
+    n = lazy(np.full(np.delete(origin, axis), ele_no, dtype=int))
```

Comment:
I would try to do this without np. I think you can just use a generator and python base functions.